### PR TITLE
Adds dictionary like argument in CLI for dict arguments in Analysis Classes

### LINKED ISSUE #38: Handle kwargs named argument in Analysis classes
Some classes have a specific named argument named `kwargs` that **does not** serve the typical kwargs functionality. Instead it is a dictionary passed to the execution method. An example of this is the `Contacts` class:

https://github.com/MDAnalysis/mdanalysis/blob/3500131284b2d91846fda9696f569bc1f6bfe741/package/MDAnalysis/analysis/contacts.py#L410

I believe we need to write an Argparse action to read a dictionary like input from the command line when a `kwargs` argument exists in the analysis class. The analysis class will know how to handle such dictionary.

### PR DESCRIPTION
**Pull request by joaomcteixeira**
Closes #38 

CLI can now read dictionaries as arguments. Dictionary syntax is parsed from JSON files, if a file is given, or from string if a string is given.

Example

```
mdacli [...] -somedictarg '{"key1": null}'
```

or

```
mdacli [...] -somedictarg my.json
```

### CODE REVIEW

File: tests/test_cli_main.py [hunk 1]
```diff
@@ -6,11 +6,17 @@
 # Released under the GNU Public Licence, v2 or any higher version
 # SPDX-License-Identifier: GPL-2.0-or-later
 """Test mdacli."""
+import argparse
+import os
+from json.decoder import JSONDecodeError
 
 import pytest
 
+from mdacli import libcli
```
Comment (PicoCentauri) [inline tests/test_cli_main.py:15]: Did you forget to add this file?

**Review by PicoCentauri — CHANGES_REQUESTED** [review]
This looks good, but did you maybe forgot including a file?

File: tests/test_cli_main.py (hunk 1 above)
Comment (joaomcteixeira) [reply tests/test_cli_main.py:15]: :drop:

File: src/mdacli/libcli.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+"""Functionalities that support the creation of the command interfaces."""
```
Comment (PicoCentauri) [inline src/mdacli/libcli.py:1]: Should we add our header here?

**Review by PicoCentauri — COMMENTED** [review]
I like the new argument. However, we should take to not confuse with our filenmaes. Now we have `utils.py` and `libcli.py` whats the what should belong where. Maybe move the `convert_str_time` method to libcli and rename utils to `docparser.py`?

**Comment by joaomcteixeira** [comment]
Yes, you are right. `libcli` might not be the best name. It is the name I use in my projects. I normally use that file to accommodate all functionalities that refer to construct the argparse interface. If we follow my conventions I would keep  `convert_str_time` outside `libcli`. But, we can also move what is inside `libcli` to another file.

File: src/mdacli/libcli.py (hunk 2 above)
Comment (joaomcteixeira) [reply src/mdacli/libcli.py:1]: If we keep the file at the end, yes.

**Comment by joaomcteixeira** [comment]
I run the tests locally in my machine and they passed. CI tests are still dependent on MDAnalysis version in Anaconda.

**Comment by PicoCentauri** [comment]
> I run the tests locally in my machine and they passed. CI tests are still dependent on MDAnalysis version in Anaconda.

Can't we use the new beta on pypi: https://pypi.org/project/MDAnalysis/2.0.0b0/ and https://pypi.org/project/MDAnalysisTests/2.0.0b0/?

**Comment by PicoCentauri** [comment]
> Yes, you are right. `libcli` might not be the best name. It is the name I use in my projects. I normally use that file to accommodate all functionalities that refer to construct the argparse interface. If we follow my conventions I would keep `convert_str_time` outside `libcli`. But, we can also move what is inside `libcli` to another file.

No make sense to have it there. We can also leave it for now and think about when we do the next restructuring iteration.

**Comment by joaomcteixeira** [comment]
> > I run the tests locally in my machine and they passed. CI tests are still dependent on MDAnalysis version in Anaconda.
> 
> Can't we use the new beta on pypi: https://pypi.org/project/MDAnalysis/2.0.0b0/ and https://pypi.org/project/MDAnalysisTests/2.0.0b0/?

Yes it is possible. For now we can merge if you agree with the implementation. I will PR later for the MDA 2.0 beta. Sounds good?

**Pull request merged**